components/apps.js

```diff
@@ -649,33 +647,137 @@ SteamUser.prototype.getOwnedDepots = function(excludeSharedLicenses) {
  * Check if this account owns a depot. Only works if enablePicsCache option is enabled and appOwnershipCached event
  * has been emitted.
  * @param {int} depotid
- * @param {boolean} [excludeSharedLicenses=false] - Pass true to exclude licenses that we have through family sharing
+ * @param {object|function} options - Options for what counts for ownership, or a custom filter function
  * @returns {boolean}
  */
-SteamUser.prototype.ownsDepot = function(depotid, excludeSharedLicenses) {
-	return this.getOwnedDepots(excludeSharedLicenses).indexOf(parseInt(depotid, 10)) != -1;
+SteamUser.prototype.ownsDepot = function(depotid, options) {
+	return this.getOwnedDepots(options).indexOf(parseInt(depotid, 10)) != -1;
 };
 
+
+
 /**
- * Returns an array of package IDs this account owns. Only works if enablePicsCache option is enabled and appOwnershipCached event
- * has been emitted.
- * @param {boolean} [excludeSharedLicenses=false] - Pass true to exclude licenses that we have through family sharing
+ * Returns an array of licenses this account owns.
  * @returns {int[]}
  */
-SteamUser.prototype.getOwnedPackages = function(excludeSharedLicenses) {
-	if (this.steamID.type != SteamID.Type.ANON_USER && !this.licenses) {
+SteamUser.prototype.getOwnedLicenses = function() {
+	if (!this.licenses) {
 		throw new Error("We don't have our license list yet.");
 	}
 
+	return this.licenses;
+}
+
+/**
+ * Returns an array of package IDs this account owns (different from owned licenses). The filter options
+ * only work, if enablePicsCache option is enabled and appOwnershipCached event has been emitted.
+ * @param {object|function} options - Options for what counts for ownership, or a custom filter function
+ * @returns {int[]}
+ */
+SteamUser.prototype.getOwnedPackages = function(options) {
+	// We're anonymous
 	if (this.steamID.type == SteamID.Type.ANON_USER) {
 		return [17906];
 	}
 
 	// We're an individual user
-	let packages = this.licenses;
-	if (excludeSharedLicenses) {
-		packages = packages.filter(license => license.owner_id == this.steamID.accountid);
+	let packages = this.getOwnedLicenses();
+
+	// Adds support for the previous syntax
+	if (!options) {
+		// Slight deviation of old method: don't return expired licenses
+		return packages.filter(license => license.flags !== SteamUser.ELicenseFlags.Expired).map(license => license.package_id);
 	}
+	// If options is a boolean, we asssume it's excludeSharedLicenses
+	if (typeof options === 'boolean') {
+		let excludeSharedLicenses = options;
```

Comment:
Could you instead just reassign `options = {excludeShared: true}` so we don't duplicate logic? Then it'll just fall through to the next if.

Reply:
> Could you instead just reassign `options = {excludeShared: options}` so we don't duplicate logic? Then it'll just fall through to the next if.

Funnily enough, that is how it was 5 commits ago, before I start implementing your feedback. I would be down for it, but I don't think it's possible to do this, and keep support for the old behavior at the same time. I think it's fine how it is now.

Reply:
I meant after checking `typeof options === 'boolean'`, then reassign options. I could have been clearer.

Reply:
If I understand it correctly, it will only move the logic down a bit, but I can do it.

Reply:
Yes, it would basically act the same as if the user has passed `{excludeShared: true}`, meaning that filter logic doesn't need to be duplicated.